neurodamus/core/coreneuron_configuration.py

```diff
@@ -242,25 +287,28 @@ def write_spike_population(self, population_name, population_offset=None):
 
     @run_only_rank0
     def write_spike_filename(self, filename):
-        report_config = Path(self.output_root) / self.report_config_file
+        report_config = Path(self.report_config_file_save)
         report_config.parent.mkdir(parents=True, exist_ok=True)
         with report_config.open("a") as fp:
             fp.write(filename)
             fp.write("\n")
 
-    def psolve_core(self, save_path=None, restore_path=None, coreneuron_direct_mode=False):
+    def psolve_core(self, coreneuron_direct_mode=False):
         from neuron import coreneuron
 
         from . import NeurodamusCore as Nd
 
         Nd.cvode.cache_efficient(1)
         coreneuron.enable = True
         coreneuron.file_mode = not coreneuron_direct_mode
-        coreneuron.sim_config = f"{self.output_root}/{self.sim_config_file}"
-        if save_path:
-            coreneuron.save_path = save_path
-        if restore_path:
-            coreneuron.restore_path = restore_path
+        coreneuron.sim_config = f"{self.sim_config_file}"
+        # set save_path only if the user explicitly asked with --save
+        # in this way we do not create 1_2.dat and time.dat if not needed
+        if SimConfig.save:
+            coreneuron.save_path = self.save_path
+        if self.restore_path:
```

Comment:
why not `if SimConfig.restore` ? To be consistent with `SimConfig.save`

Reply:
done